File: 0242-valid-anagram-有效的字母异位词/0242.cpp
#include <map>

class Solution {
    std::map<char, int> table;
public:
    bool isAnagram(string s, string t) {
        for (const auto &c: s) {
            if (table.find(c) == table.end()) {
                table[c] = 1;
            } else {
                table[c]++;
            }
        }
        for (const auto &c: t) {
            if (table.find(c) == table.end()) {
                return false;
            } else {
                table[c]--;
            }
        }
        for (const auto &n: table) {
            if (n.second != 0) {
                return false;
            }
        }
        return true;
    }
};

File: 0242-valid-anagram-有效的字母异位词/0242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        table = {}
        for c in s:
            if c in table:
                table[c] += 1
            else:
                table[c] = 1
        for c in t:
            if c in table:
                table[c] -= 1
                if table[c] < 0:
                    return False
            else:
                return False
        for k, v in table.items():
            if v > 0:
                return False
        return True